app/plugin.go

```diff
@@ -486,9 +486,11 @@ func (a *App) DisablePlugin(id string) *model.AppError {
 }
 
 func (ch *Channels) disablePlugin(id string) *model.AppError {
-	if collectionType, ok := ch.collectionTypes[id]; ok {
+	if collectionTypes, ok := ch.collectionTypes[id]; ok {
+		for _, collectionType := range collectionTypes {
+			delete(ch.topicTypes, collectionType)
+		}
 		delete(ch.collectionTypes, id)
-		delete(ch.topicTypes, collectionType)
 	}
```

Comment:
Given `nil` semantics and `delete` for a non-existent key being a no-op, could we simplify to just:
```
	for _, collectionType := range ch.collectionTypes[id] {
		delete(ch.topicTypes, collectionType)
	}
	delete(ch.collectionTypes, id)
```